### ISSUE TITLE
make trust_dns_proto Message::update_counts public

### ISSUE THREAD

**Issue by Sherlock-Holo** [bluejekyll/trust-dns#1155]
**Is your feature request related to a problem? Please describe.**
I am trying to use trust_dns_proto to write a dns server, instead of using `DnsRequest`, I am using the `Message` directly but it can't use `set_query_count` because `Message` only implements `Deref` and can return the `&Header`. If `Message` can return a `&mut Header` or implement `DerefMut` this problem can be fixed.

**Describe the solution you'd like**
`Message` can return a `&mut Header` or implement `DerefMut` 

**Describe alternatives you've considered**
No good idea, only can use `message.queries().len()` to get the query count

**Comment by fluxxu** [comment]
This field will be updated automatically before the `Message` been encoded to its binary form:

https://github.com/bluejekyll/trust-dns/blob/main/crates/proto/src/op/message.rs#L88
https://github.com/bluejekyll/trust-dns/blob/main/crates/proto/src/op/message.rs#L793
https://github.com/bluejekyll/trust-dns/blob/main/crates/proto/src/op/message.rs#L799

**Comment by bluejekyll** [comment]
Exactly. I'd be a little concerned about exposing that as that the logic in this path is not obvious. We serialize the Message, and leave space in the buffer to go back and fill in the header. This is because of the way that we handle things like signing the message for SIG0.
